traffic_control/ingress.go

```diff
@@ -0,0 +1,270 @@
+package istio_less_ingress_controller
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+type ingressEntry struct {
+	ingress *v1beta1.Ingress
+	path    v1beta1.HTTPIngressPath
+}
+
+type ingressCache struct {
+	mutex sync.Mutex
+
+	hostToIngressEntry map[string][]ingressEntry
```

Comment:
`hostToIngressEntry` caches all potential hosts in-cluster and maps to a list of imgress entries (each for a predcitor), so the istio-less ingress controller seems to be a cluster scope deployment? however, each `Inference` service creates it, it is expected?